solr/core/src/test/org/apache/solr/TestCpuTimeSearch.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import org.apache.lucene.tests.util.LuceneTestCase;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.params.ShardParams;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.security.AllowListUrlChecker;
+import org.apache.solr.util.SolrJettyTestRule;
+import org.apache.solr.util.ThreadStats;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+
+public class TestCpuTimeSearch extends SolrTestCaseJ4 {
+
+  @ClassRule public static final SolrJettyTestRule solrRule = new SolrJettyTestRule();
+
+  private static SolrClient collection1;
+  private static SolrClient collection2;
+  private static String shard1;
+  private static String shard2;
+
+  @BeforeClass
+  public static void setupSolr() throws Exception {
+    System.setProperty(ThreadStats.ENABLE_CPU_TIME, "true");
+    System.setProperty(AllowListUrlChecker.DISABLE_URL_ALLOW_LIST, "true");
+
+    Path configSet = createTempDir("configSet");
+    copyMinConf(configSet.toFile());
+    solrRule.startSolr(LuceneTestCase.createTempDir());
+
+    solrRule.newCollection("collection1").withConfigSet(configSet.toString()).create();
+    solrRule.newCollection("collection2").withConfigSet(configSet.toString()).create();
+    collection1 = solrRule.getSolrClient("collection1");
+    collection2 = solrRule.getSolrClient("collection2");
+
+    String urlCollection1 = solrRule.getBaseUrl() + "/" + "collection1";
+    String urlCollection2 = solrRule.getBaseUrl() + "/" + "collection2";
+    shard1 = urlCollection1.replaceAll("https?://", "");
+    shard2 = urlCollection2.replaceAll("https?://", "");
+
+    SolrInputDocument doc = new SolrInputDocument();
+    doc.setField("id", "1");
+    doc.setField("subject", "batman");
+    doc.setField("title", "foo bar");
+    collection1.add(doc);
+    collection1.commit();
+
+    doc.setField("id", "2");
+    doc.setField("subject", "superman");
+    collection2.add(doc);
+    collection2.commit();
+  }
+
+  @AfterClass
+  public static void cleanup() throws Exception {
+    if (null != collection1) {
+      collection1.close();
+      collection1 = null;
+    }
+    if (null != collection2) {
+      collection2.close();
+      collection2 = null;
+    }
```

Comment:
the test rule will auto-close these things; you can simply set to null.  Even that could be done in a test infrastructure test rule with reflection but we don't have one that does that (would be cool!)